### PR TITLE
MAT-5714: Update ui order for population criteria for both QI-core and QDM

### PR DESCRIPTION
**Pull request by mcmcphillips**
## MADiE PR

Jira Ticket: [MAT-5714](https://jira.cms.gov/browse/MAT-5714)
(Optional) Related Tickets:

### All Submissions
* [x] This PR has the JIRA linked.
* [x] Required tests are included.
* [x] No extemporaneous files are included (i.e Complied files or testing results).
* [x] This PR is merging into the **correct branch**.
* [x] All Documentation needed for this PR is Complete (or noted in a TODO or other Ticket).
* [x] Any breaking changes or failing automations are noted by placing a comment on this PR.

### DevSecOps
If there is a question if this PR has a security or infrastructure impact, please contact the Security or DevOps engineer assigned to this project to discuss it further.

* [ ] This PR has NO significant security impact (i.e Changing auth methods, Adding a new user type, Adding a required but vulnerable package).
* [ ] All CDN/Web dependencies are hosted internally (i.e MADiE-Root Repo).

### Reviewers
By Approving this PR you are attesting to the following:

*  Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose.
*  The tests appropriately test the new code, including edge cases.
*  If you have any concerns they are brought up either to the developer assigned, security engineer, or leads.

### CODE REVIEW

**Review by sb-cecilialiu — DISMISSED** [review]

File: src/components/editMeasure/populationCriteria/groups/QICore/QICoreMeasureGroups.tsx [hunk 1]
```diff
@@ -175,6 +176,16 @@ const MeasureGroups = (props: MeasureGroupProps) => {
   >([]);
   const { updateMeasure } = measureStore;
   const [measure, setMeasure] = useState<Measure>(measureStore.state);
+  // one additional step to check the measure is proportion, and if so verify the sort order for existing measures for a ui enhancement
```
Comment (RohitKandimalla) [inline src/components/editMeasure/populationCriteria/groups/QICore/QICoreMeasureGroups.tsx]: Is this comment appropriate here ?

File: src/components/editMeasure/populationCriteria/groups/QICore/QICoreMeasureGroups.tsx [hunk 2]
```diff
@@ -175,6 +176,16 @@ const MeasureGroups = (props: MeasureGroupProps) => {
...
+  const modifyMeasure = (measure) => {
+    setMeasure(measure);
+  };
+  useEffect(() => {
+    const subscription = measureStore.subscribe(modifyMeasure);
```
Comment (RohitKandimalla) [inline src/components/editMeasure/populationCriteria/groups/QICore/QICoreMeasureGroups.tsx]: Can we do `measureStore.subscribe(setMeasure);` ?

File: src/components/editMeasure/populationCriteria/groups/QICore/QICoreMeasureGroups.tsx [hunk 3]
```diff
@@ -925,6 +945,7 @@ const MeasureGroups = (props: MeasureGroupProps) => {
                     render={(arrayHelpers) => (
                       <div id="populations-content">
                         {formik.values.populations?.map((population, index) => {
+                          // console.log('formik.values', formik.values.populations)
```
Comment (RohitKandimalla) [inline src/components/editMeasure/populationCriteria/groups/QICore/QICoreMeasureGroups.tsx:948]: Could be removed.

File: src/components/editMeasure/populationCriteria/groups/QICore/QICoreMeasureGroups.tsx [hunk 4]
```diff
@@ -263,6 +274,15 @@ const MeasureGroups = (props: MeasureGroupProps) => {
 
   useEffect(() => {
     if (measure?.groups && measure?.groups[measureGroupNumber]) {
+      // manually sort group populations in existing groups on retrieval
+      const group = measure?.groups[measureGroupNumber];
+      group.populations.sort((a, b) => {
+        return (
+          allPopulations.findIndex((all) => all.name === a.name) -
+          allPopulations.findIndex((all) => all.name === b.name)
```
Comment (RohitKandimalla) [inline src/components/editMeasure/populationCriteria/groups/QICore/QICoreMeasureGroups.tsx:278]: Do we have to do this for QDM MeasureGroups as well ?

File: src/components/editMeasure/populationCriteria/groups/QICore/QICoreMeasureGroups.tsx (hunk 3 above)
Comment (RohitKandimalla) [inline src/components/editMeasure/populationCriteria/groups/QICore/QICoreMeasureGroups.tsx]:
```suggestion

```

**Review by RohitKandimalla — DISMISSED** [review]

File: src/components/editMeasure/populationCriteria/groups/QICore/QICoreMeasureGroups.tsx (hunk 4 above)
Comment (mcmcphillips) [reply src/components/editMeasure/populationCriteria/groups/QICore/QICoreMeasureGroups.tsx:278]: Surprisingly not. For whatever reason just sorting the array order helpers fixed it for QDM.

**Review by jkotanchik-SB — APPROVED** [review]

**Review by chubert-sb — APPROVED** [review]

**Pull request merged**